src/shaders/StarFieldProgram.js

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2014 United States Government as represented by the Administrator of the
+ * National Aeronautics and Space Administration. All Rights Reserved.
+ */
+/**
+ * @exports StarFieldProgram
+ */
+define([
+        '../error/ArgumentError',
+        '../shaders/GpuProgram',
+        '../util/Logger'
+    ],
+    function (ArgumentError,
+              GpuProgram,
+              Logger) {
+        "use strict";
+
+        /**
+         * Constructs a new program.
+         * Initializes, compiles and links this GLSL program with the source code for its vertex and fragment shaders.
+         * <p>
+         * This method creates WebGL shaders for the program's shader sources and attaches them to a new GLSL program.
+         * This method then compiles the shaders and then links the program if compilation is successful.
+         * Use the bind method to make the program current during rendering.
+         *
+         * @alias StarFieldProgram
+         * @constructor
+         * @augments GpuProgram
+         * @classdesc StarFieldProgram is a GLSL program that draws points representing stars.
+         * @param {WebGLRenderingContext} gl The current WebGL context.
+         * @throws {ArgumentError} If the shaders cannot be compiled, or linking of the compiled shaders into a program
+         * fails.
+         */
+        var StarFieldProgram = function (gl) {
+            var vertexShaderSource =
+                    //.x = declination
+                    //.y = right ascension
+                    //.z = point size
+                    //.w = magnitude
+                    'attribute vec4 vertexPoint;\n' +
+
+                    'uniform mat4 mvpMatrix;\n' +
+                    //number of days (positive or negative) since Greenwich noon, Terrestrial Time,
+                    // on 1 January 2000 (J2000.0)
+                    'uniform float numDays;\n' +
+                    'uniform vec2 magnitudeRange;\n' +
+
+                    'varying float magnitudeWeight;\n' +
+
+                    //normalizes an angle between 0.0 and 359.0
+                    'float normalizeAngle(float angle) {\n' +
+                    '   float angleDivisions = angle / 360.0;\n' +
+                    '   return 360.0 * (angleDivisions - floor(angleDivisions));\n' +
+                    '}\n' +
+
```

Comment:
@strikerM, could this be replaced by the built-in GLSL step function?
http://docs.gl/el3/step